### DISCUSSION TITLE
Server specs for 300M users/month

### DISCUSSION THREAD

**Discussion by Tony-MF** [plausible/analytics#2464] [Self-Hosted Support] [answered]
Hi,

We run a high-traffic business with thin margins. That's why the paid version is not sustainable for us. So I'd like to give a shot to the self-hosted version and support the project if it works for us.

Currently, we have about 5M users / day. I'd like to be ready for 10M users/day. 

Does anyone have a recommendation:

1. How much RAM and threads I will need approximately for 10M/day? 
2. What DB is less HW consuming, Postgres or Clickhouse?
3. Anything else I should be aware of?

Thanks in advance

**Comment by ukutaht** [comment, marked as answer]
I think for high-traffic sites our prices are very competitive, because the biggest resource hog is Clickhouse read path. If you run your own instance, you'll be paying for the hardware 100% of the time even when you actually run queries <1% of the time. On the shared Plausible.Cloud instance you'll always have access to our large Clickhouse servers for queries.

More specific specs I would recommend below:

Postgres will be fine with very basic specs. It will be OK with 2vCPUs and even just 1GB of ram.

For the app server I would recommend starting with at least 16vCPUs. App server is not very RAM intensive at the moment.

The biggest problem will be Clickhouse. It will handle ingestion fine even with lower specs but to get any kind of reasonable query performance from it you'd want to use at least 64GB ram and 32vCPUs on the low end. Our nodes are bigger than that and it still takes ~5 seconds to aggregate large queries.

Aggregating so many data points live requires some beefy HW. If you're comfortable managing bare metal, that's the best option.
